es-app/src/FileData.cpp

```diff
@@ -134,12 +126,27 @@ std::vector<FileData*> FileData::getFilesRecursive(unsigned int typeMask, bool d
 	return out;
 }
 
+std::string FileData::getKey() {
+	if (mIsVirtual)
+		return getLongKey();
+	else
+		return getShortKey();
+}
+
+FileData* FileData::getMetadataSourceFileData()
```

Comment:
From a 'client' point of view, would you care if the FileData you are getting is the original source, or a copy of the source? For example, are we expecting to have changes in the copy that are not propagated through to the original? Will we maybe at somepoint need a 'regular' `get()` as well?